app/com/gu/contentapi/sanity/SSLExpiryTest.scala

```diff
@@ -0,0 +1,62 @@
+package com.gu.contentapi.sanity
+
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import scala.util.{Try, Success, Failure}
+import org.scalatest.{FlatSpec}
+import javax.net.ssl._
+import java.net.{URL}
+import org.joda.time.{Days, DateTime}
+import sun.security.x509.X509CertImpl
+
+import scala.util.Try
+
+class SSLExpiryTest extends FlatSpec with ScalaFutures with IntegrationPatience {
+
+  "SSL Certificates" should "be more than 30 days from expiry" taggedAs (LowPriorityTest) in {
+    handleException {
+      val hosts = Seq(
+        Config.host,
+        Config.hostPublicSecure,
+        Config.previewHost,
+        Config.writeHost,
+        Config.writePreviewHost
+      )
+
+      val secureHosts = hosts map {
+        _.replaceAll("http://", "https://")
+      }
+
+      for (host <- secureHosts) {
+
+        val url = new URL(host)
+        val conn = url.openConnection().asInstanceOf[HttpsURLConnection]
+        conn.setHostnameVerifier(new HostnameVerifier {
+          override def verify(hostnameVerifier: String, sslSession: SSLSession) = true
+        }
+        )
+
+        val certsTry = Try {
+          conn.connect()
+          conn.getServerCertificates
+        }
+        conn.disconnect()
+
+        certsTry match {
+          case Success(certs) =>
+            certs.headOption.map { cert =>
+              val x = cert.asInstanceOf[X509CertImpl]
```

Comment:
Not a big problem (I'd say it's fine to merge as-is), but before calling `asInstanceOf` I would usually have an assertion to check that the thing is actually what I am trying to cast it to.

```
certs.headOption.map { cert =>
  cert shouldBe a[X509CertImpl]
  val x = cert.asInstanceOf[X509CertImpl]
  ...
}
```